pkg/build/nodeimage/internal/kube/builder_remote.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+type remoteBuilder struct {
+	version string
+	arch    string
+	logger  log.Logger
+}
+
+var _ Builder = &remoteBuilder{}
+
+func NewRemoteBuilder(logger log.Logger, version, arch string) (Builder, error) {
+	return &remoteBuilder{
+		version: version,
+		arch:    arch,
+		logger:  logger,
+	}, nil
+}
+
+// Build implements Bits.Build
+func (b *remoteBuilder) Build() (Bits, error) {
+	url := "https://dl.k8s.io/" + b.version + "/kubernetes-server-linux-" + b.arch + ".tar.gz"
```

Comment:
Do we forsee supporting downstream release hosts? Maybe we punt that to an env override or a totally different `--type` later. That could get messy to support (e.g. we have binaries versus images, and the releases hosting layout is sort of a weak API that is more reasonable when we're just consuming upstream)

Reply:
covered this use case, we just let them specify direct url(s) - https://github.com/kubernetes-sigs/kind/pull/3614#issuecomment-2121474989